### PR TITLE
Show localised error when user rejects transaction in MetaMask

### PR DESCRIPTION
**Pull request by ahoppen**
At the moment, when the user rejects a transaction from within MetaMask, a stack trace of the error within the MetaMask extension is displayed. This gives very little value to the user as to what actually happened.

Checking for `setTxStatusRejected` in the error message seems to be the safest way to check if the transaction was rejected from within MetaMask that works in all browsers since Firefox clips the start of the error message.

I have translated the new message to German but left the other languages untouched since I speak neither of them. I don't know what the standard policy for that is and if they would be required for the PR to be accepted.

### CODE REVIEW

**Comment by ahoppen** [comment]
The entire check is pretty tailor-made for MetaMask, since I'm checking for `setTxStatusRejected`, which is a MetaMask-internal function that occurs in the error's stack trace. I can remove the reference to MetaMask, but don't think this will make the check any more general.

File: app/public/i18n/de.json [hunk 1]
```diff
@@ -1 +1 @@
... Schnittstelle","contractInformation":"Vertragsinformationen","addedContract":"Erstelle Vertrag","readFromContract":"Vertrag auslesen","writeToContract":"In Vertrag schreiben","pickContract":"Wähle einen Vertrag","pickFunction":"Wähle eine Funktion","noContract":"<p>Fügen Sie den Quellcode eines Vertrags hier ein. Beispiele finden Sie unter <a href=\"http://ethereum.org\" target=\"_blank\">ethereum.org</a>: </p> <ul><li><a href=\"http://ethereum.org/token#the-code\" target=\"_blank\">Token erzeugen</a></li><li><a href=\"http://ethereum.org/crowdsale#the-code\" target=\"_blank\">Crowdsale starten</a></li><li><a href=\"http://ethereum.org/dao#the-code\" target=\"_blank\">Blockchain-Organisation erzeugen</a></li></ul>","buttons":{"showContractInfo":"Zeige Vertragsinformationen","hideContractInfo":"Schließe Vertragsinformationen","watchContractEvents":"Beobachte Events","execute":"Ausführen","yes":"Ja"},"error":{"compile":"Quellcode konnte nicht kompiliert werden.","noDataProvided":"Keine Daten gefunden: Wählen Sie einen Vertrag aus dem Menü","invalidAddress":"Vertrag nicht hinzugefügt, ungültige Adresse.","jsonInterfaceParseError":"JSON Schnittstelle konnte nicht gelesen werden."}}}}}
```
Comment (Hasanyeach) [inline app/public/i18n/de.json]: Thanks

**Comment by evertonfraga** [comment]
@ahoppen alright. should we have any other wallet with similar functionality, we make the text more general.

**Comment by Hasanyeach** [comment]
Saya tidak bisa mendownload metamask melalui smartphone 😑

**Pull request merged**